### PR TITLE
add draft of CITATION.cff

### PR DESCRIPTION
**Pull request by pdiercks**
Hi, 
I added a draft for how this repo may be cited in the [citation file format](https://github.com/citation-file-format/citation-file-format/blob/1.2.0/schema-guide.md#definitionspersonaffiliation).

### CODE REVIEW

File: CITATION.cff [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    family-names: Diercks
+    email: philipp.diercks@bam.de
+    affiliation: >-
+      Bundesanstalt für Materialforschung und
+      -prüfung (BAM)
+  - given-names: Dennis
+    family-names: Gläser
+    email: dennis.glaeser@iws.uni-stuttgart.de
+    affiliation: >-
+      Universität Stuttgart
```
Comment (dglaeser) [inline CITATION.cff:23]:
```suggestion
      University of Stuttgart
```

File: CITATION.cff [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    family-names: Unger
+    email: joerg.unger@bam.de
+    affiliation: >-
+      Bundesanstalt für Materialforschung und
+      -prüfung (BAM)
+  - given-names: Bernd
+    family-names: Flemisch
+    email: bernd.flemisch@iws.uni-stuttgart.de
+    affiliation: >-
+      Universität Stuttgart
```
Comment (dglaeser) [inline CITATION.cff:34]:
```suggestion
      University of Stuttgart
```

File: CITATION.cff (hunk 1 above)
Comment (dglaeser) [reply CITATION.cff:23]: maybe better to use the English names?

**Comment by dglaeser** [comment]
Thanks @pdiercks, just a minor comment about the English names.. Looks good otherwise. Did you use the online service that helps with cff file creation? That would ensure that the syntax is correct.

**Comment by joergfunger** [comment]
BAM wants us to always use the German name (no translation of Bundesanstalt für Materialforschung und -prüfung). I guess this is a personal decision of each institution.

**Comment by pdiercks** [comment]
Hi @dglaeser , I think I used the online service to get started (not fully sure though, I did this in July and forgot to request review).

**Comment by pdiercks** [comment]
Hi @dglaeser ,
here are some ToDo's to improve the citation file:

- [ ] add DOI
- [ ] add a suitable CFF type, see [About citation files](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-citation-files#citing-something-other-than-software)

**Comment by pdiercks** [comment]
Hi @dglaeser 
I updated the citation file. Did I miss some important information?

**Comment by dglaeser** [comment]
@pdiercks, looks good to me. What do you think about adding a validation step to the actions, though? I found this here: https://github.com/dieghernan/cff-validator

**Comment by pdiercks** [comment]
@dglaeser , the zenodo doi is https://doi.org/10.5281/zenodo.7790634 
Thanks! I'll add the cff validation step in the actions.

**Comment by pdiercks** [comment]
hi @dglaeser ,
the field "type" in the cff file is an enum and supported values are `dataset` or `software`. I set it to `dataset`, because the repo is just a collection of scripts and documentation. Do you think it is rather `software`?

This is also why I was confused about the data availability and software availability statement in the paper. Having a zenodo doi pointing to the repo does not resolve the issue that we only have *one* dataset, but no software associated with the paper?

File: .github/workflows/cff.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  pull_request:
+    branches: [ main ]
+      
+  workflow_dispatch:
+
+jobs:
+  validate-citation-cff:
+    runs-on: ubuntu-latest
+    env:
+      GITHUB_PAT: ${{ secrets.GITHUB_TOKEN }}
```
Comment (dglaeser) [inline .github/workflows/cff.yml:15]: I saw this also in the examples of the validation action, but I am wondering why it is needed? We do check out the repo before the validation step, is that action querying more stuff from our repo via the API?

Comment (dglaeser) [reply .github/workflows/cff.yml:15]: Looking at the action code (https://github.com/dieghernan/cff-validator/blob/main/action.yml), I don't see where `GITHUB_PAT` is used, but I guess it wouldn't be in the examples if it wasn't necessary... I'm just usually a bit hesitant to use secrets in the CI :)

**Review by dglaeser — APPROVED** [review]

File: .github/workflows/cff.yml (hunk 3 above)
Comment (pdiercks) [reply .github/workflows/cff.yml:15]: I was also not sure about the env variable. It seems that the action is based on R and that authentication is needed for set-up or something ...

**Pull request merged**